### PR TITLE
chore: update stuff

### PR DESCRIPTION
**Pull request by DmitriyMV**
- Bump ci script.
- Bump github.com/ProtonMail/gopenpgp/v2 to v2.7.3
- Bump github.com/grpc-ecosystem/grpc-gateway/v2 to v2.17.1
- Add `List.Iterator` method and deprecate `IteratorFromList`.
- `safe.State*` functions should accept `state.CoreState` where possible.

### CODE REVIEW

File: pkg/safe/state.go [hunk 1]
```diff
@@ -299,13 +299,20 @@ func (l *List[T]) Find(fn func(T) bool) (T, bool) {
 	return zero, false
 }
 
+// Iterator returns a new iterator over the list.
+func (l *List[T]) Iterator() ListIterator[T] {
```
Comment (frezbo) [inline pkg/safe/state.go:303]: any particular reason to deprecate, or just for easier use?

**Review by frezbo — APPROVED** [review]

File: pkg/safe/state.go (hunk 1 above)
Comment (DmitriyMV) [reply pkg/safe/state.go:303]: Just easier to use. I don't remember why I didn't do this in the first place. It looks better too!

Comment (frezbo) [reply pkg/safe/state.go:303]: yes, feels better than having to call safe.Iterator

**Pull request merged**